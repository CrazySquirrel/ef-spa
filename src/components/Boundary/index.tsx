import * as Sentry from '@sentry/browser';
import * as React from 'react';

interface Props {
}

interface State {
  error: any;
}

export class Boundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      error: null,
    };
  }

  public componentDidCatch(error: any, errorInfo: any) {
    this.setState({error});

    Sentry.withScope((scope) => {
      Object.keys(errorInfo).forEach((key) => {
        scope.setExtra(key, errorInfo[key]);
      });

      Sentry.captureException(error);
    });
  }

  public render() {
    if (this.state.error) {
      // render fallback UI
      return (
          <a onClick={() => Sentry.showReportDialog()}>Отправить отчет</a>
      );
    } else {
      // when there's not an error, render children untouched
      return this.props.children;
    }
  }
}
