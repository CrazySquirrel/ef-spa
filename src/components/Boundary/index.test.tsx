import * as React from 'react';
import {shallow} from 'enzyme';

import Boundary from './index';

describe('Boundary', () => {
  it('render', () => {
    const result = shallow(<Boundary/>);

    expect(result).toMatchSnapshot();
  });
});
