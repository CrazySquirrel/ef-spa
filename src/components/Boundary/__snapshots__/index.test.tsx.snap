version https://git-lfs.github.com/spec/v1
oid sha256:0d803fba998f1ccc04309b3fe7790f150d74fa50165e60bd7950d58f2e6a0283
size 992
